pkg/utils/opsqueue.go

```diff
@@ -15,33 +15,34 @@
 package utils
 
 import (
+	"math/bits"
 	"sync"
 
-	"github.com/livekit/protocol/logger"
+	"github.com/gammazero/deque"
+	"github.com/livekit/protocol/utils"
 )
 
 type OpsQueue struct {
-	logger logger.Logger
-	name   string
-	size   int
+	name        string
+	flushOnStop bool
 
-	lock      sync.RWMutex
-	ops       chan func()
+	lock      sync.Mutex
+	ops       deque.Deque[func()]
+	wake      chan struct{}
 	isStarted bool
+	doneChan  chan struct{}
 	isStopped bool
 }
 
-func NewOpsQueue(logger logger.Logger, name string, size int) *OpsQueue {
-	return &OpsQueue{
-		logger: logger,
-		name:   name,
-		size:   size,
-		ops:    make(chan func(), size),
+func NewOpsQueue(name string, minSize uint, flushOnStop bool) *OpsQueue {
+	oq := &OpsQueue{
+		name:        name,
+		flushOnStop: flushOnStop,
+		wake:        make(chan struct{}, 1),
+		doneChan:    make(chan struct{}),
```

Comment:
this should probably be set up in `Start` since the queue can be restarted

Reply:
Restart is not supported. It checks for `isStarted` and returns if already started.  I had this in `Start` and then changed it. I will move it into `Start()` method as it is more correct like you mentioned even if restart is not supported.

Reply:
oh, if it can't restart nvm

Reply:
ok, will not change it. Made it a bit awkward. Will not change for now. If restarting is supported, we can change it then.